union.go

```diff
@@ -0,0 +1,67 @@
+package rwvfs
+
+import (
+	"fmt"
+	"io"
+	"os"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
```

Comment:
How precisely do you want it to be a "union"? Is it okay if some features are not supported?

For example, is it a union of distinct filesystems, or is overlap between filesystems allowed? If allowed, does it gracefully merge contents of folders with same path?

I ask because I know it currently does not handle every single possible scenario, but perhaps that's okay. It would be nice to see the limitations documented though, as that would clear up this question. On the other hand, if having correct union of overlapping filesystems is the goal, that should be fixed (I can provide a failing test case).